File: backend/app/schemas/history.py
from __future__ import annotations

from enum import Enum
from typing import List, Optional

from pydantic import BaseModel


class HistoryRange(str, Enum):
    M15 = "15m"
    H1 = "1h"
    H24 = "24h"
    D7 = "7d"


class HistoricalMetricSample(BaseModel):
    timestamp_ms: int
    cpu_usage_percent: float
    cpu_temperature_celsius: Optional[float]
    memory_usage_percent: float
    network_download_bytes_per_second: Optional[float]
    network_upload_bytes_per_second: Optional[float]


class HistoricalMetricsResponse(BaseModel):
    range: HistoryRange
    start_timestamp_ms: int
    end_timestamp_ms: int
    samples: List[HistoricalMetricSample]

File: frontend/src/types/history.ts
export type HistoricalRange = '15m' | '1h' | '24h' | '7d';

export const HISTORICAL_RANGES: HistoricalRange[] = ['15m', '1h', '24h', '7d'];

export interface HistoricalMetricSample {
  timestamp_ms: number;
  cpu_usage_percent: number;
  cpu_temperature_celsius: number | null;
  memory_usage_percent: number;
  network_download_bytes_per_second: number | null;
  network_upload_bytes_per_second: number | null;
}

export interface HistoricalMetricsResponse {
  range: HistoricalRange;
  start_timestamp_ms: number;
  end_timestamp_ms: number;
  samples: HistoricalMetricSample[];
}
